SCons/Script/SConsOptions.py

```diff
@@ -709,6 +710,11 @@ def opt_duplicate(option, opt, value, parser):
                   action="help",
                   help="Print this message and exit.")
 
+    op.add_option('--hash-format',
```

Comment:
Can we add this as a limited list with all viable options?

Reply:
In the original design I left it open-ended so as Python adds more factories to hashlib, people will get them for free. So I just want to be clear that if we make this change, the solution will be more limited than it could be, but the advantage is that it would be more closely vetted by us (i.e. closed-ended list means that we can cover everything with tests).

So then, the question becomes: what list do we want to use? Definitely MD5, SHA1, and SHA256. Should I also include SHA512 and blake2b?

Reply:
which hashes are currently supported in each supported version of python (3.5->3.9)?
Is it reasonable to extract this from the version of python being used to create the list?

Assuming that (for now), that would resolve your use case?

Reply:
According to https://docs.python.org/3/library/hashlib.html 3.5 supports MD5, SHA1, SHA224, SHA256, SHA384, and SHA512. 3.6 adds blake2b/blake2s and SHA3 algorithms.

That said, I don't know if/how that applies to pypi.

Reply:
I'd be fine with introspecting the python standard lib hashes to create the permitted list for now.  If someone really needs/wants more, then we can address that later.

Reply:
@grossag - would this limitation be sufficient to address your use case(s)?

Reply:
That would be `hashlib.algorithms_available` and `hashlib.algorithms_guaranteed`, depending on your bent

Reply:
should be added that md5 will appear in the lists even on no-md5 "FIPS builds" - a topic which brought us here previously.

Reply:
What happens if you try to use md5 on a FIPS build. Does it throw an exception?
Maybe SCons.Utils should have a get_available_hashes() function?

Reply:
I haven't been able to reproduce the specific environment - I have a "FIPS CentOS", but md5 still works there, so there's more to it somehow. All I know is in issue #2949 (and I haven't re-read that recently)

Reply:
Here's the values from sugguested list above in my py3.9 build
>>> hashlib.algorithms_available
{'sha512', 'sha512_256', 'md4', 'sha3_512', 'shake_256', 'blake2s', 'sha256', 'sha1', 'sm3', 'md5-sha1', 'sha384', 'sha512_224', 'ripemd160', 'sha3_384', 'shake_128', 'blake2b', 'sha3_256', 'sha224', 'mdc2', 'md5', 'whirlpool', 'sha3_224'}
>>> hashlib.algorithms_guaranteed
{'sha384', 'sha3_256', 'sha3_512', 'sha224', 'sha3_224', 'shake_256', 'blake2s', 'sha256', 'md5', 'sha1', 'sha3_384', 'shake_128', 'blake2b', 'sha512'}